trieste/space.py

```diff
@@ -241,11 +247,48 @@ def __contains__(self, value: TensorType) -> bool | TensorType:
         return tf.reduce_all(value >= self._lower) and tf.reduce_all(value <= self._upper)
 
     def sample(self, num_samples: int) -> TensorType:
+        """
+        Sampling points randomly
+
+        :param num_samples: The number of points to sample from this search space.
+        :return: ``num_samples`` i.i.d. random points, sampled uniformly, and without replacement,
+            from this search space.
+        """
```

Comment:
Please add in tf.debugging.assert_positive(num_samples) for all the three sample methods along with tests that the methods fail with invalid sample size. For example see line 38 in https://github.com/secondmind-labs/trieste/blob/develop/tests/unit/acquisition/test_sampler.py

Reply:
I used .assert_non_negative, since we also accept 0 as num_samples